### ISSUE TITLE
Strongly constrained date entry?

### ISSUE THREAD

**Issue by metaperl** [reahl/reahl#233]
When we look at the current DateField in reahl [1] we see a `parse_input` method that has free-form input. Ie., you can input fuzzy values and the parser converts this into Python objects. 

A more common way to enter dates is via a date picker or with rigid fields for month date and time. Is there any possibility for datefields to be entered via a datepicker? There are [many boostrap examples of datepickers available](https://www.google.com/search?q=bootstrap+calendar+entry&rlz=1C1GCEA_enUS781US781&oq=bootstrap+calendar+entry&aqs=chrome..69i57.8063j0j4&sourceid=chrome&ie=UTF-8).

[1] https://github.com/reahl/reahl/blob/master/reahl-component/reahl/component/modelinterface.py#L1246

**Comment by IwanVosloo** [comment]
Yes, this is something we had planned to do. Will give it some more priority now that someone's asked....
